### PR TITLE
Add ListOptions to GetCommit and CompareCommits to support pagination

### PR DESCRIPTION
**Pull request by Saaarah**

### CODE REVIEW

File: github/repos_commits.go
Comment (gmlewis) [inline github/repos_commits.go:239]: Please add a blank line after 239 for consistency with the rest of the repo.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thanks, @Saaarah.

Please make sure to run unit tests before submitting PRs.
You will need to modify the unit tests since the API has changed with this PR.

Please check out surrounding tests to see how pagination is also tested.  Thanks.

**Comment by Saaarah** [comment]
> Thanks, @Saaarah.
> 
> Please make sure to run unit tests before submitting PRs.
> You will need to modify the unit tests since the API has changed with this PR.
> 
> Please check out surrounding tests to see how pagination is also tested. Thanks.

Hi Glenn, thanks for reviewing my merge request and I've added the new parameters in the two API's unit tests. Thanks again!

**Comment by gmlewis** [comment]
Hi @Saaarah - it looks like there are some calls to these APIs in the examples, if you could please take a look and fix them.

Here is the error:

```
Run go vet ./...
# github.com/google/go-github/v37/example/commitpr
vet: example/commitpr/main.go:128:97: too few arguments in call to client.Repositories.GetCommit
Error: Process completed with exit code 2.
```

**Comment by Saaarah** [comment]
> Hi @Saaarah - it looks like there are some calls to these APIs in the examples, if you could please take a look and fix them.
> 
> Here is the error:
> 
> ```
> Run go vet ./...
> # github.com/google/go-github/v37/example/commitpr
> vet: example/commitpr/main.go:128:97: too few arguments in call to client.Repositories.GetCommit
> Error: Process completed with exit code 2.
> ```

Hi Glenn, I've fixed the example. Sorry for not updating all affected code, it'll be nice to have a small section in readme.md to ask contributors to run all necessary tests prior to creating pull request, which will be really helpful for first timer contributors.

**Comment by gmlewis** [comment]
> it'll be nice to have a small section in readme.md to ask contributors to run all necessary tests prior to creating pull request, which will be really helpful for first timer contributors.

This is already covered in our [CONTRIBUTING.md](https://github.com/google/go-github/blob/master/CONTRIBUTING.md) doc:

```
  5. Please run:
     * `go generate github.com/google/go-github/...`
     * `go test github.com/google/go-github/...`
     * `go vet github.com/google/go-github/...`
```

**Comment by gmlewis** [comment]
It now looks like `gofmt` needs to be run on the example code:

```
Run diff -u <(echo -n) <(gofmt -d -s .)
--- /dev/fd/63	2021-07-12 15:58:15.735418376 +0000
+++ /dev/fd/62	2021-07-12 15:58:15.735418376 +0000
@@ -0,0 +1,12 @@
+diff -u example/commitpr/main.go.orig example/commitpr/main.go
+--- example/commitpr/main.go.orig	2021-07-12 15:58:16.035434090 +0000
++++ example/commitpr/main.go	2021-07-12 15:58:16.035434090 +0000
+@@ -127,7 +127,7 @@
+ 	// Get the parent commit to attach the commit to.
+ 	opt := &github.ListOptions{
+ 		PerPage: 10,
+-		Page: 1,
++		Page:    1,
+ 	}
+ 	parent, _, err := client.Repositories.GetCommit(ctx, *sourceOwner, *sourceRepo, *ref.Object.SHA, opt)
+ 	if err != nil {
Error: Process completed with exit code 1.
```

**Comment by Saaarah** [comment]
> > it'll be nice to have a small section in readme.md to ask contributors to run all necessary tests prior to creating pull request, which will be really helpful for first timer contributors.
> 
> This is already covered in our [CONTRIBUTING.md](https://github.com/google/go-github/blob/master/CONTRIBUTING.md) doc:
> 
> ```
>   5. Please run:
>      * `go generate github.com/google/go-github/...`
>      * `go test github.com/google/go-github/...`
>      * `go vet github.com/google/go-github/...`
> ```

Ah thanks for pointing out! I've formatted the file and run all the commands listed above. Thanks Glenn!

File: example/commitpr/main.go
Comment (gmlewis) [inline example/commitpr/main.go:132]: Since the focus of this example is not pagination, let's please just get rid of lines 128-131 and change line 132 to pass in `nil` instead of `opt`.

**Review by gmlewis — CHANGES_REQUESTED** [review]

**Review by gmlewis — APPROVED** [review]
Thank you, @Saaarah !
LGTM.

Awaiting second LGTM before merging.

**Review by Parker77 — APPROVED** [review]
LGTM.

**Pull request merged**
